test/functional/tests/memory/test_memory_metadata_consumption.py

```diff
@@ -0,0 +1,195 @@
+        pass_criteria:
+          - Successful CAS creation and cores addition.
+          - Memory consumption within limits.
+    """
+    with TestRun.step("Prepare drives and partitions for core devices."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(50, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks["core"]
+        parts = [Size(3200, Unit.MebiByte)] * cores_count
+        core_disk.create_partitions(parts)
+        core_devices = core_disk.partitions
+
+    with TestRun.step("Start cache with configuration and add all core devices."):
+        free_mem_before_start = get_mem_available()
+
+        cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+
+        for dev in core_devices:
+            cache.add_core(dev)
+
+    with TestRun.step("Measure allocates size of metadata."):
+        used_memory = get_module_mem_footprint(cas_cache_module)
+        TestRun.LOGGER.info(f"Memory allocated: {used_memory}")
+
+    with TestRun.step(
+        "Compare allocated size with cache usage statistics " "and required consumption."
+    ):
+        free_mem_after_start = get_mem_available()
+
+        metadata_max_size = calculate_expected_metadata_footprint(cache.size, cache_line_size)
+        TestRun.LOGGER.info(f"Metadata max size: {metadata_max_size}")
+
+        validate_memory_consumption(cache, metadata_max_size, used_memory)
+
+        memory_delta = free_mem_before_start - free_mem_after_start
+        ram_change_status = is_memory_within_limit(
+            "Measured increment of RAM usage after start", metadata_max_size, memory_delta
+        )
+
+        if not ram_change_status:
+            TestRun.LOGGER.error(f"Memory measurements completed, found problems.")
+
+
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_memory_metadata_consumption_full_cache(cache_line_size):
+    """
+        title: Test for CAS RAM consumption for metadata for cache usage 100%.
+        description: |
+          Validate CAS RAM consumption for metadata depends on start cache line size
+          for cache usage 100% (with 5% tolerance) where cache mode Write-Back is used.
+        pass_criteria:
+          - Successful CAS creation and cores addition.
+          - Filling all cache space successfully.
+          - Memory consumption within limits.
+    """
+    with TestRun.step("Prepare drives and partitions for core devices."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(50, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks["core"]
+        parts = [Size(3200, Unit.MebiByte)] * cores_count
```

Comment:
How about making part size a cache size divided by number of cores? That way we eliminate possibility that sum of cores will be smaller than cache, which now may be achieved by unfortunate combination if independent parameters.

Reply:
ok